src/main/java/appeng/tile/powersink/AEBasePoweredTile.java

```diff
@@ -19,7 +19,308 @@
 package appeng.tile.powersink;
 
 
-public abstract class AEBasePoweredTile extends RedstoneFlux
+import java.util.EnumSet;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.energy.IEnergyStorage;
+
+import appeng.api.config.AccessRestriction;
+import appeng.api.config.Actionable;
+import appeng.api.config.PowerMultiplier;
+import appeng.api.config.PowerUnits;
+import appeng.api.networking.energy.IAEPowerStorage;
+import appeng.api.networking.events.MENetworkPowerStorage.PowerEventType;
+import appeng.capabilities.Capabilities;
+import appeng.integration.Integrations;
+import appeng.integration.abstraction.IC2PowerSink;
+import appeng.tile.AEBaseInvTile;
+import appeng.tile.TileEvent;
+import appeng.tile.events.TileEventType;
+
+
+public abstract class AEBasePoweredTile extends AEBaseInvTile implements IAEPowerStorage, IExternalPowerSink
 {
 
+	// values that determine general function, are set by inheriting classes if
+	// needed. These should generally remain static.
+	private double internalMaxPower = 10000;
+	private boolean internalPublicPowerStorage = false;
+	private AccessRestriction internalPowerFlow = AccessRestriction.READ_WRITE;
+	// the current power buffer.
+	private double internalCurrentPower = 0;
+	private EnumSet<EnumFacing> internalPowerSides = EnumSet.allOf( EnumFacing.class );
+	private final IEnergyStorage forgeEnergyAdapter;
+	private Object teslaEnergyAdapter;
+
+	private IC2PowerSink ic2Sink;
+
+	public AEBasePoweredTile()
+	{
+		this.forgeEnergyAdapter = new ForgeEnergyAdapter( this );
+		if( Capabilities.TESLA_CONSUMER != null )
+		{
+			this.teslaEnergyAdapter = new TeslaEnergyAdapter( this );
+		}
+		this.ic2Sink = Integrations.ic2().createPowerSink( this, this );
+		this.ic2Sink.setValidFaces( this.internalPowerSides );
+	}
+
+	protected EnumSet<EnumFacing> getPowerSides()
+	{
+		return this.internalPowerSides.clone();
+	}
+
+	protected void setPowerSides( final EnumSet<EnumFacing> sides )
+	{
+		this.internalPowerSides = sides;
+		this.ic2Sink.setValidFaces( sides );
+		// trigger re-calc!
+	}
+
+	@TileEvent( TileEventType.WORLD_NBT_WRITE )
+	public void writeToNBT_AERootPoweredTile( final NBTTagCompound data )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00100)